### DISCUSSION TITLE
OpenJDK updates?

### DISCUSSION THREAD

**Discussion by loeffelpan** [antlafarge/jdownloader#29] [Q&A] [answered]
I found your image and considering to use it in favor of my local installation on ubuntu.
But I'm wondering what about updates of OpenJDK8? I think you have to rebuild the image to apply those and it's not something I can do.

And what about OpenJDK11 or newer? There are packages in alpine linux and my local instance is running fine with OpenJDK11 for years now.

**Comment by antlafarge** [comment, marked as answer]
Hello,

Yes I used OpenJDK 8 because it was simpler to find a working package at the date I created the project, but anyway it would be a good idea to upgrade to a newer LTS version (11 or 17). I will look at this when I will have some time.
Thank you for your feedback !

**Reply by loeffelpan** [reply]
Thanks for looking at this.
Please give feedback when this is done.

**Reply by antlafarge** [reply]
Yes there is no emergency because the image works well :)
But I will upgrade to the last java LTS as soon as I have some time.

**Comment by antlafarge** [comment]
Hello, I created new images which uses OpenJDK 17.
The `latest`, `ubuntu` and `alpine` tags point always to version OpenJDK 8 by waiting for more tests and feedback.
You can look and try this new version by changing the tag to `openjdk17`, `ubuntu-openjdk17`, or `alpine-openjdk17`:
https://hub.docker.com/repository/docker/antlafarge/jdownloader/tags?page=1&ordering=last_updated&name=openjdk17

**Comment by loeffelpan** [comment]
Sounds good. Will take a look at this soon. Thanks.

As OpenJDK is always security relevant:
Do you see any way to configure unattended upgrades of OpenJDK (at least for Ubuntu via apt should be possible)?

**Comment by antlafarge** [comment]
If security is really important, you should rebuild the image yourself regularly (you have a build.md file available in the github sources).
On the other hand it is about a sandboxed container so I do not think that it is so critical.
For my part I try to update the image several times a year.

**Reply by loeffelpan** [reply]
As docker containers run with root permissions, it could be important as well.
But maybe you're right. I will use it as you build it.

**Reply by antlafarge** [reply]
I recommand to start the docker container as your user in the documentation, but you should take care about directories owners and access rights to do that.
Then you will have multiple layers of security.
And you can use the `users` group (gid `100`) to manage easily access rights to the directories for multiple users.
https://github.com/antlafarge/jdownloader#setup

**Reply by loeffelpan** [reply]
I already use that with uid and gui 33 (www-data) as I'm in an environment with Nextcloud.
But I guess that this only means that this user will be used inside the cotainer. I still need "root" to start the container as for any other docker image.

But nevermind. Maybe I'm a bit paranoid.

**Reply by antlafarge** [reply]
No problem let me know if you have any problems with the image =]

**Reply by antlafarge** [reply]
Now the latest version uses openjdk21, and I added a github action to build the image automatically each month (to get security upgrades from ubuntu/alpine and openjdk).
